test/common/cdp.py

```diff
@@ -137,10 +159,45 @@ def find_cdp_port(self):
 
     def get_browser_path(self):
         if self._browser_path is None:
-            self._browser_path = browser_path()
+            self._browser_path = browser_path(self.browser)
 
         return self._browser_path
 
+    def browser_cmd(self, cdp_port, env):
+        exe = self.get_browser_path()
+        if not exe:
+            raise SystemError(self.browser + " is not installed")
+
+        if self.browser == "chromium":
+            return [exe, "--headless", "--disable-gpu", "--no-sandbox", "--disable-setuid-sandbox",
+                    "--disable-namespace-sandbox", "--disable-seccomp-filter-sandbox",
+                    "--disable-sandbox-denial-logging", "--disable-pushstate-throttle",
+                    "--window-size=1920x1200", "--remote-debugging-port=%i" % cdp_port, "about:blank"]
+        elif self.browser == "firefox":
+            subprocess.Popen(["firefox", "--headless", "--no-remote", "-CreateProfile", "blank"], env=env).communicate()
+            p = subprocess.Popen("ls ~/.mozilla/firefox", shell=True, stdout=subprocess.PIPE, env=env)
+            in_profile = p.communicate()[0].decode("utf-8").split("\n")
+            userjs = "~/.mozilla/firefox/"
+            handler = "~/.mozilla/firefox/"
+            for item in in_profile:
+                if item.endswith("blank"):
+                    handler = os.path.join(handler, item, "handlers.json")
+                    userjs = os.path.join(userjs, item, "user.js")
+                    break
+            prefs = """
+            user_pref("remote.enabled", true);
+            user_pref("datareporting.policy.dataSubmissionEnabled", false);
+            user_pref("toolkit.telemetry.reportingpolicy.firstRun", false);
+            user_pref("dom.disable_beforeunload", true);
+            user_pref("browser.download.dir", "{0}");
+            user_pref("browser.download.folderList", 2);
+            """.format(self.download_dir)
+
+            subprocess.Popen("echo '{0}' > {1}".format(prefs, userjs), shell=True, env=env).communicate()
+            subprocess.Popen("echo '{\"defaultHandlersVersion\":{\"en-US\":4},\"mimeTypes\":{\"application/xz\":{\"action\":0,\"extensions\":[\"xz\"]}}}' > " + handler,
+                             shell=True, env=env).communicate()
```

Comment:
It's easier and much more robust to just do this in Python. Something like
```py
with open(os.path.join(sef._browser_home, "blank", ""user.js"), "w") as f:
    f.write("blabla")
```

For handlers.json you can pick whether you want to use `json.dump(object, f)` or write the JSON manually (not much difference for such small objects)